### PR TITLE
Convert QR tests to testsets

### PR DESCRIPTION
**Pull request by kshyatt**
I checked and all the `if eltya != Int` tests still don't work if you remove the restriction.

### CODE REVIEW

File: test/linalg/qr.jl [hunk 1]
```diff
@@ -20,150 +19,152 @@ a2img  = randn(n,n)/2
...
-debug && println("\ntype of a: ", eltya, " type of b: ", eltyb, "\n")
-debug && println("QR decomposition (without pivoting)")
+        @testset "QR decomposition of a Number" begin
+            α = rand(eltyb)
+            aα = fill(α,1,1)
+            @test qrfact(α)[:Q]*qrfact(α)[:R] ≈ qrfact(aα)[:Q]*qrfact(aα)[:R]
+            @test abs(qrfact(α)[:Q][1,1]) ≈ one(eltyb)
+        end
+
         for i = 1:2
```
Comment (andreasnoack) [inline test/linalg/qr.jl]: Maybe a bit clearer if the two loop are rewritten as
```julia
for (a, b) in ((a, b),
               (view(a, 1:n-1, 1:n-1), view(b, 1:n-1, 1:n-1)))
```

**Review by andreasnoack — APPROVED** [review]

File: test/linalg/qr.jl [hunk 2]
```diff
@@ -20,150 +19,152 @@ a2img  = randn(n,n)/2
...
 
-# Because transpose(x) == x
-@test_throws ErrorException transpose(qrfact(randn(3,3)))
-@test_throws ErrorException ctranspose(qrfact(randn(3,3)))
-@test_throws ErrorException transpose(qrfact(randn(3,3), Val(false)))
-@test_throws ErrorException ctranspose(qrfact(randn(3,3), Val(false)))
-@test_throws ErrorException transpose(qrfact(big.(randn(3,3))))
-@test_throws ErrorException ctranspose(qrfact(big.(randn(3,3))))
+@testset "transpose errors" begin
+    # Because transpose(x) == x
```
Comment (tkelman) [inline test/linalg/qr.jl]: is this comment outdated?

Comment (kshyatt) [inline test/linalg/qr.jl]: cc @andyferris

Comment (andreasnoack) [inline test/linalg/qr.jl]: Yes. We don't have the fallback anymore so the comment is outdated. We can probably also remove the methods in `factorizations.jl` that used to be exercised here and change the error tests to `MethodError`.

**Comment by kshyatt** [comment]
@andreasnoack apparently not! :p I'll try to fix this later.

File: test/linalg/qr.jl [hunk 3]
```diff
@@ -20,28 +19,29 @@ a2img  = randn(n,n)/2
...
-        α = rand(eltyb)
-        aα = fill(α,1,1)
-        @test qrfact(α)[:Q]*qrfact(α)[:R] ≈ qrfact(aα)[:Q]*qrfact(aα)[:R]
-        @test abs(qrfact(α)[:Q][1,1]) ≈ one(eltyb)
         tab = promote_type(eltya,eltyb)
 
-debug && println("\ntype of a: ", eltya, " type of b: ", eltyb, "\n")
-debug && println("QR decomposition (without pivoting)")
-        for i = 1:2
-            let a = i == 1 ? a : view(a, 1:n - 1, 1:n - 1), b = i == 1 ? b : view(b, 1:n - 1), n = i == 1 ? n : n - 1
```
Comment (fredrikekre) [inline test/linalg/qr.jl]: Failure looks to be something with the `view`s here

**Review by fredrikekre — APPROVED** [review]
looks good otherwise

**Review by tkelman — APPROVED** [review]
lgtm, do squash on merge

**Pull request merged**